app/org/sagebionetworks/bridge/services/AuthenticationServiceImpl.java

```diff
@@ -112,28 +113,27 @@ public UserSession getSession(String sessionToken) {
 
     @Override
     public UserSession signIn(Study study, SignIn signIn) throws ConsentRequiredException, EntityNotFoundException {
+
         checkNotNull(study, "Study cannot be null");
         checkNotNull(signIn, "Sign in cannot be null");
-
         Validate.entityThrowingException(signInValidator, signIn);
+
         String lockId = null;
         try {
-            lockId = lockDao.acquireLock(SignIn.class, signIn.getUsername());
+            lockId = lockDao.acquireLock(SignIn.class,
+                    study.getIdentifier() + RedisKey.SEPARATOR + signIn.getUsername());
```

Comment:
The default lock timeout is 3 minutes. It's possible that another process acquired but failed to release the lock, which would cause the lock to sit around for a very long time. I would recommend setting the explicitly setting lock timeout timeout to something more reasonable, something in the 1-5 sec range.

I'd also recommend making the same changes for signUp.